service/src/main/java/bio/terra/catalog/service/dataset/DatasetAccessLevel.java

```diff
@@ -0,0 +1,29 @@
+package bio.terra.catalog.service.dataset;
+
+import bio.terra.catalog.iam.SamAction;
+import java.util.EnumSet;
+import java.util.Set;
+
+public enum DatasetAccessLevel {
+  OWNER("owner", EnumSet.allOf(SamAction.class)),
+  READER("reader", EnumSet.of(SamAction.READ_ANY_METADATA)),
+  DISCOVERER("discoverer", EnumSet.of(SamAction.READ_ANY_METADATA));
+
+  // The UUI depends on this fields contents. Changing these will require a change to the UI.
```

Comment:
```suggestion
  // The UI depends on this fields contents. Changing these will require a change to the UI.
```

micro-nit: misspelled?